includes/payment-methods/class-wc-stripe-express-checkout-element.php

```diff
@@ -114,149 +111,10 @@ public function init() {
-	 *
-	 * @return void
-	 */
-	private function setup_custom_checkout_data() {
-		// Register our express checkout data as extended data, which will hold
-		// custom checkout fielddata if present.
-		$extend_schema = StoreApi::container()->get( ExtendSchema::class );
-		$extend_schema->register_endpoint_data(
-			[
-				'endpoint'        => CheckoutSchema::IDENTIFIER,
-				'namespace'       => 'wc-stripe/express-checkout',
-				'schema_callback' => [ $this, 'get_custom_checkout_data_schema' ],
-			]
-		);
-
-		// Update order based on extended data.
-		add_action(
-			'woocommerce_store_api_checkout_update_order_from_request',
-			[ $this, 'process_custom_checkout_data' ],
-			10,
-			2
-		);
-	}
-
-	/**
-	 * Allow third-party to validate and add custom checkout data to
-	 * express checkout orders for stores using classic, shortcode-based checkout.
-	 *
-	 * @param WC_Order $order The order to add custom checkout data to.
-	 * @param WP_REST_Request $request The request object.
-	 * @return void
-	 */
-	public function process_custom_checkout_data( $order, $request ) {
-		$custom_checkout_data = $this->get_custom_checkout_data_from_request( $request );
-		if ( empty( $custom_checkout_data ) ) {
-			return;
-		}
-
-		$errors = new WP_Error();
-		/**
-		 * Allow third-party plugins to validate custom checkout data for express checkout orders.
-		 *
-		 * To be used as a stand-in for the `woocommerce_after_checkout_validation` action.
-		 *
-		 * @since 9.6.0
-		 *
-		 * @param array $custom_checkout_data The custom checkout data.
-		 * @param WP_Error $errors The WP_Error object, for adding errors when validation fails.
-		 */
-		do_action( 'wc_stripe_express_checkout_after_checkout_validation', $custom_checkout_data, $errors );
-
-		if ( $errors->has_errors() ) {
-			$error_messages = implode( "\n", $errors->get_error_messages() );
-			throw new WC_Data_Exception( 'wc_stripe_express_checkout_invalid_data', $error_messages );
-		}
-
-		/**
-		 * Allow third-party plugins to add custom checkout data for express checkout orders.
-		 *
-		 * To be used as a stand-in for the `woocommerce_checkout_update_order_meta` action.
-		 *
-		 * @since 9.6.0
-		 *
-		 * @param integer $order_id The order ID.
-		 * @param array $custom_checkout_data The custom checkout data.
-		 */
-		do_action( 'wc_stripe_express_checkout_update_order_meta', $order->get_id(), $custom_checkout_data );
-	}
-
-	/**
-	 * Get custom checkout data from the request object.
-	 *
-	 * To support custom fields in express checkout and classic checkout,
-	 * we pass custom data as extended data, i.e. under extensions.
-	 *
-	 * @param WP_REST_Request $request The request object.
-	 * @return array Custom checkout data.
-	 */
-	private function get_custom_checkout_data_from_request( $request ) {
-		$extensions = $request->get_param( 'extensions' );
-		if ( empty( $extensions ) ) {
-			return [];
+		if ( apply_filters( 'wc_stripe_ece_enable_custom_checkout_fields_support', false ) ) {
```

Comment:
```suggestion
                 // Document the filter details here
                 $enable_custom_checkout_fields_support = apply_filters( 'wc_stripe_ece_enable_custom_checkout_fields_support', false );
		if ( $enable_custom_checkout_fields_support ) {
```

I think, it would be better if we document the filter here.